plugins/org.eclipse.fordiac.ide.deployment.opcua/src/org/eclipse/fordiac/ide/deployment/opcua/OPCUADeploymentExecutor.java

```diff
@@ -572,20 +574,83 @@ public List<org.eclipse.fordiac.ide.deployment.devResponse.Resource> queryResour
 
 	@Override
 	public Response queryFBType(final FBTypeEntry entry) throws DeploymentException {
-		// TODO implement when the API and infrastructure is fully tested
-		throw new DeploymentException("Query FB Type not yet supported!"); //$NON-NLS-1$
+		final CallMethodRequest request = new CallMethodRequest(Constants.MGMT_NODE, Constants.QUERY_FB_TYPE_NODE,
+				new Variant[] {});
+		try {
+			final String message = MessageFormat.format(Constants.QUERY_FB_TYPE, getTypeNameWithHash(entry));
```

Comment:
To set an input argument, a `new Variant()` needs to be added to the empty Variant[] array.

```suggestion
		try {
		        final String hashedFBType = getTypeNameWithHash(entry);
		        final CallMethodRequest request = new CallMethodRequest(Constants.MGMT_NODE, Constants.QUERY_FB_TYPE_NODE,	new Variant[] { new Variant(hashedFBType) });
			final String message = MessageFormat.format(Constants.QUERY_FB_TYPE, hashedFBType);
```